woocommerce/woocommerce
# Enhancement: Update account creation settings wording

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Implements account settings UI changes from @nikkivias outlined in https://github.com/woocommerce/woocommerce/issues/48533

This PR does not change existing functionality, it only rewords existing options and some minor UI elements on the settings page. Supporting changes include:

- Ability to render a visible legend above checkbox fields
- Ability to mark inputs disabled, and have the parent TD element styled with opacity if disabled

The javascript which toggles the disabled state of the account creation options does not use jQuery and is specific to this settings page.

Closes #48533

This documentation will require updating: https://woocommerce.com/document/configuring-woocommerce-settings/accounts-and-privacy/



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WooCommerce > Settings > Accounts and privacy
2. Ensure the settings render, there are no visual bugs, and you can toggle things on and off with settings saved correctly
3. Toggle off "During checkout" and "On "My account" page". Ensure account creation options are greyed out. 
4. Toggle one option on and ensure account creation options are restored.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
